semrel.go

```diff
@@ -211,84 +214,117 @@ func (listRefsQuery) vars(repo *Repository, perPage int, cursor string) map[stri
 // GetLatestRelease returns the latest release that matches the given version
 // range or prerelease. If neither is set, it returns the latest non-prerelease.
 // If verRange is unset but prerelease is set, it returns the newer of either
-// the latest non-prerelease or the latest matching prerelease. If verRange is
-// set but prerelease is unset, it returns the latest matching version for
-// verRange. If both are set, strange things may happen. It will take the latest
-// non-prerelease or matching pre-release, and use its SHA with the verRange
-// version, resulting in a return value that may not actually exist in the repo.
+// the latest non-prerelease or the latest matching prerelease.
+//
+// verRange currently must not be set, as it is no longer supported, and this
+// method will return an error if it is used.
+//
+// What it _used_ to do is: If verRange is set but prerelease is unset, it
+// returns the latest matching version for verRange. If both are set, strange
+// things may happen. It will take the latest non-prerelease or matching
+// pre-release, and use its SHA with the verRange version, resulting in a return
+// value that may not actually exist in the repo.
 func (repo *Repository) GetLatestRelease(verRange string, prerelease string) (*Release, error) {
-	var lastRelease *Release
-	var cursor string
-	var verRangeConstraint *semver.Constraints
 	if verRange != "" {
-		var err error
-		if verRangeConstraint, err = semver.NewConstraint(verRange); err != nil {
-			return nil, err
-		}
+		return nil, fmt.Errorf("version range constraints for maintained release are no longer supported")
 	}
 
-REFS:
-	for {
-		// this will do a reverse chronological sort, which SHOULD be in semrel
-		// order for at least the "main" and pre-release sequences we're hunting
-		// for.
-		var q listRefsQuery
-		err := repo.GQLClient.Query(repo.Ctx, &q, q.vars(repo, 100, cursor))
-		if err != nil {
-			return nil, fmt.Errorf("failed github tags query: %w", err)
-		}
-		for _, r := range q.Repository.Refs.Nodes {
-			version, err := semver.NewVersion(strings.TrimPrefix(string(r.Name), "refs/tags/"))
+	// run multiple searches in parallel:
+	// 1. a general search with no query filter to find the latest normal release (no pre-release component)
+	// 2. a filtered search for the given pre-release if any
+	// 3. ??? what to do with verRange?
+
+	var eg errgroup.Group
+	// locate main release
+	var lastMainRelease *Release
+	eg.Go(func() error {
```

Comment:
Is using this errgroup stuff to speed this up?

Reply:
This allows running the two in parallel with a convenient way to track any error(s) they return